stats/stats.py

```diff
@@ -64,122 +64,112 @@ def __init__(self, queryset, result_type, modifier=None, granulations=None):
-            }
-            new_document['values'] = data
+                # foreach granulation field verify if it has granulation parameters or if they
+                # are correct
+                for granulation_field, granulation_parameters in values['granulations'].iteritems():
+                    self._validate_granulation_parameters(granulation_field, granulation_parameters,
+                                                          values['granulations'])
 
-            stats_list.append(new_document)
+        return True
 
-        return stats_list
-
-    def _numbers_to_strings(self, argument):
-        switcher = {
-            1: "Jan",
-            2: "Feb",
-            3: "Mar",
-            4: "Apr",
-            5: "May",
-            6: "Jun",
-            7: "Jul",
-            8: "Aug",
-            9: "Sep",
-            10: "Oct",
-            11: "Nov",
-            12: "Dec",
-        }
-        return switcher.get(argument, "none")
+    # iterates the granulated_stats dictionary and creates the stats_list
+    def _create_stats_list(self, granulated_stats, stats_list, granulation_field,
+                           granulation_arguments):
+        for key in sorted(granulated_stats.keys()):
+            data = dict()
+            data['currency'] = key.currency
+            if 'additional_granulation_field' in granulation_arguments:
+                data['customer_name'] = key.name
+            if 'granulation_field' in granulation_arguments:
+                data[granulation_field] = key.granulation
+            data['values'] = []
+            for subscription in sorted(granulated_stats[key]):
+                data['values'].append(subscription)
+            stats_list.append(data)
 
-    def _get_time_granulations(self, document, granulation_arguments, granulation_field):
-        if 'time_granulation_interval' in granulation_arguments \
-                and 'time_granulation_interval' is not None:
-            if granulation_arguments['time_granulation_interval'] == 'year':
-                return str(getattr(document, granulation_field).year)
-            elif granulation_arguments['time_granulation_interval'] == 'month':
-                month = self._numbers_to_strings(getattr(document, granulation_field).month)
-                return str(getattr(document, granulation_field).year) + ' ' + month
-            elif granulation_arguments['time_granulation_interval'] == 'day':
-                day = self._numbers_to_strings(
-                    getattr(document, granulation_field).month) + ' ' + str(
-                    getattr(document, granulation_field).day)
-                return str(getattr(document, granulation_field).year) + ' ' + day
+    # search the elements that correspond with the granulation tuple key given
+    def _search_element(self, tuple_key, granulated_stats, granulation_arguments, customer_name,
+                        amount, id, additional_field):
```

Comment:
You don't need to receive the `granulated_stats` list here. Just return the element and append it to the granulated_stats one level up in the call tree.